src/node/waveshaper.rs

```diff
@@ -349,18 +350,19 @@ impl WaveShaperRenderer {
     }
 
     #[inline]
-    fn no_process(&self, input: &AudioRenderQuantum, output: &mut AudioRenderQuantum) {
+    fn pass_through(&self, input: &AudioRenderQuantum, output: &mut AudioRenderQuantum) {
         for (i_data, o_data) in input.channels().iter().zip(output.channels_mut()) {
-            for (&i, o) in i_data.iter().zip(o_data.iter_mut()) {
-                *o = i;
-            }
+            o_data.copy_from_slice(&i_data[..]);
         }
     }
 
     #[inline]
-    fn process_none(&self, input: &AudioRenderQuantum, output: &mut AudioRenderQuantum) {
+    fn process_1x(&self, input: &AudioRenderQuantum, output: &mut AudioRenderQuantum) {
+        // this does nothing...
         for (i_data, o_data) in input.channels().iter().zip(output.channels_mut()) {
```

Comment:
Not related to your changes, but I think this is dangerous. We should not make assumption about the shape of the output buffers. They are typically reused every render quantum but they are initialized as a single channel of silence. This means if the input buffer has 2 channels, we're actually only processing 1 of them.

Probably best to use the strategy of the GainNode - make a copy of input and mutate in place
```rust
        // single input/output node
        let input = &inputs[0];
        let output = &mut outputs[0];
        let gain_values = params.get(&self.gain);

        *output = input.clone();

        output.modify_channels(|channel| {
            channel
                .iter_mut()
                .zip(gain_values.iter())
                .for_each(|(value, g)| *value *= g)
        });
```

Or do you think we should reshape the output in the render graph before handing them to the process function, to avoid these types of problems?

Reply:
Yes you are right, I will fix that

> Or do you think we should reshape the output in the render graph before handing them to the process function, to avoid these types of problems?

No strong opinion on that point, maybe it would be a bit more clear and avoid calls  to `output.set_number_of_channels`, but at contrary maybe that would require that the `Graph` have too much knowledge of the nodes (for example would it make sens for the `StereoPannerNode` ?).